portal/src/util/misc.ts

```diff
@@ -0,0 +1,25 @@
+export function getListFromCommaSeparatedString(
+  commaSeparatedString: string,
+  optionEnabled: boolean = true,
+  useDefaultList: boolean = false,
+  defaultList: string[] = []
+): string[] {
+  if (!optionEnabled) {
+    return [];
+  }
+  const list = commaSeparatedString.split(",");
+  const sanitizedList = list.map((item) => item.trim()).filter(Boolean);
+  return useDefaultList ? [...sanitizedList, ...defaultList] : sanitizedList;
+}
+
+export function setFieldIfListNonEmpty(
```

Comment:
I prefer we just put these two functions in the original caller file first. When we have more users, we can move it here.